src/documents/CoreAPI.md

```diff
@@ -0,0 +1,635 @@
+* Returns: `{Array|Object}` A shallow clone of `toFilter` with the supplied keys removed
+
+
+### fluid.keys(object)
+
+Return the keys in the supplied object as an array. Note that this will return keys found in the prototype chain as well as "own properties", unlike the builtin [`Object.keys()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys)
+
+* `object {Object}` The object to have its keys listed
+* Returns: `{Array}` An array holding the keys of this object.
+
+
+### fluid.values(object)
+
+Return the keys in the supplied object as an array. This will return values found in the prototype chain as well as those attached to "own properties".
+
+* `object {Object}` The object to have its values listed
+* Returns: `{Array}` An array holding the values of this object.
+
+
+### fluid.arrayToHash(array)
+    
+Converts an array into an object whose keys are the elements of the array, each with the value `true`
+* `array {Array of String}` The array to be converted to a hash
+* Returns: `{Object}` An object with value `true` for each key taken from a member of `array`
+
+
+### fluid.hashToArray(hash, keyName[, func])
+
+Converts a hash into an array by hoisting out the object's keys into an array element via the supplied String `key`, and then transforming the elements via an optional further function `func`.
+
+* `hash {Array}` The hash to be transformed into an array
+* `keyName {String}` The name within the resulting array elements which will receive the string which used to form the element's key in `hash`
+* `func {Function (newElement {Object), oldElement {Object}, key {String}) →  Object }` [optional] receives `(newElement, oldElement, key)` where `newElement` is the freshly cloned element (a fresh `Object obj` which just has `obj[keyName] = key`), 
+`oldElement` is the original hash's element, and `key` is the key of the element in the hash. The function returns the final element that will be added to the array - it may ignore `newElement` if it pleases and make an unrelated
+return. If the function makes no return, `newElement` will be used. If the function is not supplied, the old element is simply deep-cloned onto the new element (same effect as transform `fluid.transforms.objectToArray`).
+
+
+### fluid.flatten(array)
+
+Converts an array consisting of a mixture of arrays and non-arrays into the concatenation of any inner arrays 
+with the non-array elements. The original array *** will not be modified ***. See description of [mapcat](http://martinfowler.com/articles/collection-pipeline/flat-map.html) or `flat-map`.
+
+* `array {Array}` The array to be flattened
+* Returns: `{Array}` The flattened array.
+
+
+## Sorting and searching
+
+### fluid.find(source, func[, deflt])
+
+Scan through a list or hash of objects, terminating on the first member which
+matches a predicate function. The return is the return value from the predicate.
```

Comment:
I think the word "matches" here implies an equivalency that is inaccurate; perhaps "satisfies" instead? (same for 'matches' in the description of 'deflt')